Quality socioeconomics (#311)
elixir-europe/infectious-diseases-toolkit: quality-control/socioeconomic-data.md

### Timeliness
* **Quality dimension**: Currency and Relevance
* Socio-economic data related to infectious diseases must be current and promptly collected to reflect the rapidly changing dynamics of disease spread and its socio-economic impacts, especially during an outbreak. Timely data enables timely intervention and policy response.

### Missing values
* **Quality dimension**: Completeness and population coverage
* Strategies should be applied to address gaps and ensure coverage of all relevant sub-populations.

## Existing approaches

### Pilot tests

> Conduct pilot tests or preliminary surveys with a subset of the target population to ensure the survey design works effectively. This helps in identifying potential issues with question clarity, relevance, and respondent understanding before full-scale data collection.

Reviewer: ```suggestion
Conduct pilot tests or preliminary surveys with a subset of the target population to ensure that the survey design works effectively. This helps to identify potential issues with the clarity, relevance, and understanding of the questions before full-scale data collection.
```

> When the data are harvested from existing databases, conduct pilot extractions to verify the data retrieval process. This involves testing the extraction protocols on a small scale to ensure that the data pulled from databases are accurate, relevant, and compatible with the study’s objectives. Pilot testing the data extraction helps identify and address any discrepancies or issues in data format, completeness, and alignment with the research questions.

Reviewer: ```suggestion
When the data are harvested from existing databases, conduct pilot extractions to verify the data retrieval process. This involves testing the extraction protocols on a small scale to ensure that the data pulled from databases are accurate, relevant, and compatible with the study’s objectives. Pilot testing the data extraction helps to identify and address any discrepancies or issues in data format, completeness, and alignment with the research questions.
```

> This helps with representativeness, assessing whether data collection methods effectively reach all segments of the population. Examining population coverage helps identify underserved or marginalized groups whose experiences may be underrepresented in the data, thus enabling more equitable decision-making.

Reviewer: ```suggestion
Ensuring that you extract the correct data from the database(s) in question helps with representativeness, as you can assess whether data collection methods effectively reach all segments of the population. Examining population coverage helps to identify under-served or marginalised groups whose experiences may be underrepresented in the data, thus enabling more equitable decision-making.
```

### Timeliness

> Implement efficient data collection protocols that ensure rapid access to and sharing of data. Real-time or near-real-time data collection and processing systems are critical for monitoring the socio-economic impacts of infectious diseases and for timely decision-making.

Reviewer: ```suggestion
Implement efficient data collection protocols that ensure rapid access to, and sharing of, data. Real-time or near-real-time data collection and processing systems are critical for monitoring the socioeconomic impacts of infectious diseases and for timely decision making.
```

### Error and inconsistency detection

> Perform regular error detection and data cleaning processes. This includes identifying and correcting outliers, inconsistencies, and missing data to improve the overall quality and reliability of the data.

Reviewer: ```suggestion
Perform regular error detection and data cleaning processes. This includes identifying outliers, inconsistencies, and missing data, and then applying corrections accordingly, to improve the overall quality and reliability of the data.
```

> Outliers (i.e., data points that significantly differ from other observations) must be identified. It is key to make a distinction between these actual but rare events that need careful consideration and errors in data entry or measurement anomalies.

Reviewer: ```suggestion
Outliers (i.e. data points that significantly differ from other observations) must be identified. It is key to make a distinction between outliers caused by erroneous data (e.g. due to errors in data entry or measurement), and those caused by real, but perhaps rare, events, as they must be handled differently.
```

Some methods for detecting errors and inconsistencies are:

> * Data Visualization Tools:

Reviewer: ```suggestion
* Data Visualisation Tools:
```

  * Python libraries: {% tool "matplotlib" %}, {% tool "seaborn" %}
* Statistical Analysis Software:
  * {% tool "r" %}
  * Python libraries: {% tool "pandas" %}, {% tool "numpy" %}
* Develop Data Quality Assessment Framework (DQAF) or implement already existing ones
* Cross-Validation Techniques:
  * {% tool "scikit-learn" %} - Cross-validation (python)

### Data cleaning

> Data cleaning procedures are essential for ensuring the accuracy and reliability of socio-economic data. This involves identifying and rectifying errors, inconsistencies, and outliers that may arise during data collection or entry. Cleaning procedures may include deduplication to remove redundant entries, standardization of formats to ensure uniformity, and imputation techniques to address missing values.

Reviewer: ```suggestion
Data cleaning procedures are essential for ensuring the accuracy and reliability of socioeconomic data. This involves identifying and rectifying errors, inconsistencies, and outliers that may arise during data collection or entry. Cleaning procedures may include de-duplication to remove redundant entries, standardisation of formats to ensure uniformity, and imputation techniques to address missing values.
```

> Existing software can be used for data cleaning, such as python libraries (e.g., {% tool "pandas" %} or {% tool "dedupe" %}).

Reviewer: ```suggestion
Existing software, such as python libraries (e.g., {% tool "pandas" %} or {% tool "dedupe" %}), can be used for data cleaning.
```

### Interoperability
* Semantic harmonisation: Use standard codes and terminologies for variables to ensure consistency across different data sources and studies.

> * Syntactic harmonisation: Adopt standardised data formats to facilitate data integration and comparison across various datasets.

Reviewer: ```suggestion
* Syntactic harmonisation: Adopt standardised data formats to facilitate data integration and comparability across various datasets.
```

> ### Statistical tests for Data Quality

Reviewer: ```suggestion
### Statistical tests for data quality
```

Conduct statistical tests to assess data reliability and validity. Reliability tests assess the consistency of the data over time, ensuring that data trends are dependable and not subject to random fluctuations. Validity tests, on the other hand, examine the accuracy of the data in measuring what they are intended to measure. For instance, in the context of infectious diseases, validity tests might assess whether socio-economic variables accurately reflect the true impact of the disease on different demographics.

### Check completeness

> Completeness is the dimension that takes into account the number of variables in a given data model that have actual values. This involves scrutinising datasets for missing values, ensuring that all relevant variables are captured

Reviewer: ```suggestion
Completeness is the dimension that takes into account the number of variables in a given data model that have actual values. This involves scrutinising datasets for missing values, ensuring that all relevant variables are captured.
```
